include/boost/uuid/detail/random_provider_wincrypt.ipp

```diff
@@ -0,0 +1,86 @@
+/* boost uuid/detail/random_provider_wincrypt implementation
+*
+* Copyright Jens Maurer 2000
+* Copyright 2007 Andy Tompkins.
+* Copyright Steven Watanabe 2010-2011
+* Copyright 2017 James E. King, III
+*
+* Distributed under the Boost Software License, Version 1.0. (See
+* accompanying file LICENSE_1_0.txt or copy at
+* http://www.boost.org/LICENSE_1_0.txt)
+*
+* $Id$
+*/
+
+#include <boost/core/ignore_unused.hpp>
+#include <boost/winapi/crypt.hpp>
+#include <boost/winapi/get_last_error.hpp>
+#include <boost/throw_exception.hpp>
+#include <string>
+
+#if !defined(BOOST_UUID_RANDOM_PROVIDER_NO_LIB)
```

Comment:
BOOST_ALL_NO_LIB, as I already said.

Reply:
Sorry I missed that!  Isn't each library capable of having its own preprocessor macros in this area, like `BOOST_SYSTEM_NO_LIB`, `BOOST_CONFIG_NO_LIB`, `BOOST_PYTHON_NO_LIB`...?

Reply:
It is, but you have to check the global macro as well and not autolink if it's set. It's a global user switch that requests all autolinking to be disabled, anywhere.

Reply:
In my builds on Windows with msvc-14.1, BOOST_ALL_NO_LIB is set to 1 by default.  This seems odd - why would it be set on the way in?  Boost.Uuid does not set it.

Reply:
Interesting. No idea why it's set; you'll need to investigate.

Reply:
It is defined for all builds of boost, it's in boost.jam.  I will add BOOST_UUID_SOURCE and make it conditional on that as well.